# **[Scraping] Tugas Besar IFB-310 Machine Learning**
## Tujuan

Melakukan scraping review pengguna dari aplikasi **Wondr by BNI** yang tersedia di Google Play Store menggunakan **google-play-scraper**. Data yang dikumpulkan akan digunakan untuk analisis sentimen pada proyek Submission 1 kelas Belajar Fundamental Deep Learning Dicoding. Target pengambilan review adalah minimal 3.000 data dan maksimal 10.000 data.

# **1. Import Library**
Pada tahap ini, saya mengimpor beberapa pustaka (library) Python yang dibutuhkan untuk scraping komentar.

In [65]:
# Library scraping dari Google Play Store
from google_play_scraper import app, reviews_all, Sort

# Manipulasi dan analisis data
import pandas as pd
pd.options.mode.chained_assignment = None
import numpy as np

# Konfigurasi seed dan download data
seed = 0
np.random.seed(seed)
nltk.download('punkt')
nltk.download('stopwords')


[nltk_data] Downloading package punkt to
[nltk_data]     C:\Users\Administrator\AppData\Roaming\nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package stopwords to
[nltk_data]     C:\Users\Administrator\AppData\Roaming\nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


True

# **2. Informasi Aplikasi**

In [66]:
# Menampilkan informasi aplikasi Wondr by BNI dari Google Play Store
app_info = app('id.bni.wondr')

# Tampilkan nama dan deskripsi singkat aplikasi
print(f"Nama Aplikasi : {app_info['title']}")
print(f"Developer     : {app_info['developer']}")
print(f"Deskripsi     :\n{app_info['description'][:500]}...")


Nama Aplikasi : wondr by BNI
Developer     : PT Bank Negara Indonesia (Persero) Tbk.
Deskripsi     :
Looking for a financial partner to simplify your life? wondr by BNI's newest features got you covered! Everything finance-related is now easier to manage and handle! 

With wondr by BNI, you can:
🔸 Withdraw cash without card – Use Mobile Tunai feature and you are good to go!
🔸 Manage your credit card easily – Apply, activate, convert transaction into installments, turn your limit into cash, and customize your credit card to fit your lifestyle
🔸Track your assets and loans with Portfolio – Al...


# **3. Scraping Review dari Google Play Store**

In [68]:
# Fungsi scraping semua review dari aplikasi Wondr by BNI
reviews_data = reviews_all(
    'id.bni.wondr',
    lang='id',  # Bahasa Indonesia
    country='id',
    sort=Sort.NEWEST,  # Urutkan berdasarkan yang terbaru
)

# Ubah ke DataFrame
df_reviews = pd.DataFrame(reviews_data)
print(f"Jumlah review yang berhasil diambil: {len(df_reviews)}")

# Tampilkan 5 review pertama
df_reviews[['userName', 'content', 'score', 'at']].head()


Jumlah review yang berhasil diambil: 39316


userName                                            content  score  \
0  Pengguna Google                              bagus cuma agak lelet      4   
1  Pengguna Google                                       bagus mantaf      5   
2  Pengguna Google  sampe kapan gangguan sistim terus, mau daftar ...      3   
3  Pengguna Google  Transaksi cepat dan mudah berkat aplikasi ini....      5   
4  Pengguna Google                 wonder lebih bagus dari BNI mobile      5   

                   at  
0 2025-05-21 09:36:49  
1 2025-05-21 09:24:05  
2 2025-05-21 09:14:49  
3 2025-05-21 09:06:12  
4 2025-05-21 09:01:32

# **4. Menyimpan Komentar ke CSV**


In [69]:
# Simpan ke file CSV
df_reviews[['userName', 'content', 'score', 'at']].to_csv('reviews_wondr_bni.csv', index=False)
print("Data berhasil disimpan sebagai 'reviews_wondr_bni.csv'")


Data berhasil disimpan sebagai 'reviews_wondr_bni.csv'


# **5. Penutup**


## Hasil Akhir
- Total review terkumpul: 39316
- Sumber: Google Play Store - Wondr by BNI (`id.bni.wondr`)
- File disimpan: `reviews_wondr_bni.csv`
- Dataset ini akan digunakan untuk submission 1 BFDL (Analisis Sentimen).

Proses scraping selesai!
